gtsfm/evaluation/visualize_benchmark_comparison.py

```diff
@@ -0,0 +1,227 @@
+    return colorscale
+
+
+def plot_colored_table(X: List[str], Y: List[str], Z: np.ndarray) -> str:
+    """Create an annotated heatmap.
+
+    Args:
+        X: labels for each column (column names).
+        Y: labels for each row (row names).
+        Z: 2d matrix, representing percentage changes from value for a metric on the master branch.
+
+    Returns:
+        string representing html for table.
+    """
+
+    # Clip Z to -20% and +20%. The clipping is only for the color -- the text will still display the correct numbers.
+    Z_clipped = np.clip(Z, a_min=MIN_RENDERABLE_PERCENT_CHANGE, a_max=MAX_RENDERABLE_PERCENT_CHANGE)
+
+    redgreen = [RED_HEX, PALE_YELLOW_HEX, GREEN_HEX]
+    colorscale = colorscale_from_list(redgreen)
+    trace = go.Heatmap(
+        z=Z_clipped,
+        x=X,
+        y=Y,
+        colorscale=colorscale,
+        zmin=-MIN_RENDERABLE_PERCENT_CHANGE,
+        zmax=MAX_RENDERABLE_PERCENT_CHANGE,
+    )
+
+    layout = go.Layout(
+        title="Percentage Change",
+        font=Font(family="Balto, sans-serif", size=12, color="rgb(68,68,68)"),
+        showlegend=False,
+        xaxis=XAxis(title="", showgrid=True, side="top", tickangle=-45),
+        yaxis=YAxis(
+            title="",
+            autorange="reversed",
+            showgrid=True,
+        ),
+        autosize=False,
+        height=HEATMAP_HEIGHT,
+        width=HEATMAP_WIDTH,
+        margin=Margin(l=135, r=40, b=85, t=170),
+    )
+
+    fig = go.Figure(data=go.Data([trace]), layout=layout)
+
+    annotations = go.Annotations()
+
+    num_rows, num_cols = Z.shape
+    for i in range(num_rows):
+        for j in range(num_cols):
+            annotations.append(
+                Annotation(
+                    text=str(np.round(Z[i, j], 1)) + "%",
+                    x=X[j],
+                    y=Y[i],
+                    xref="x1",
+                    yref="y1",
+                    font=dict(color="rgb(25,25,25)"),
+                    showarrow=False,
+                )
+            )
+    fig["layout"].update(annotations=annotations)
+    return fig.to_html(full_html=False, include_plotlyjs="cdn")
+
+
+def generate_dashboard(curr_master_dirpath: str, new_branch_dirpath: str) -> None:
+    """Generate a dashboard showing a visual representation of the diff against master on all benchmarks.
+
+    This script expects to find the metrics in CI artifact files and saves to the main repo directory.
+    TODO(johnwlambert): read metrics from JSON, instead of from the HTML report.
+
+    Args:
+        curr_master_dirpath: path to directory containing benchmark artifacts for the master branch.
+        new_branch_dirpath: path to directory containing benchmark artifacts for a new branch.
+    """
+
+    f = open(DASHBOARD_HTML_SAVE_FPATH, mode="w")
+
+    # Write HTML headers.
+    f.write("<!DOCTYPE html>" "<html>")
+    f.write(metrics_report.get_html_header())
+
+    # loop over each table in the HTML report.
```

Comment:
nit: capitalize first letter of sentence.

Reply:
thanks, updated